### PR TITLE
Android 7.4.0 release

### PR DESCRIPTION
**Pull request by HuFangshuai**
Android 7.4.0 sample changes.

### CODE REVIEW

**Review by ManojBudumuru-vungle — APPROVED** [review]

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AppopenFragment.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private InterstitialAd interstitialAd;
+
+    public AppopenFragment() {
+        // doesn't do anything special
+    }
+
+    @Override
+    protected void destroyAd() {
+        super.destroyAd();
+    }
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AppopenFragment.java:22]: why is this needed?

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AppopenFragment.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        setTextColor(_binding.btnPlayAd, R.color.black);
+
+    }
+
+    @Override
+    public void onAdStart(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbAdStart, R.color.black);
+        _binding.btnPlayAd.setEnabled(false);
+        setTextColor(_binding.btnPlayAd, R.color.lightGray);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/AppopenFragment.java:87]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    private VungleBannerView bannerAd;
+
+    public InlineFragment() {
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return super.onCreateView(inflater, container, savedInstanceState);
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:29]: why is this needed?

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        super.onViewCreated(view, savedInstanceState);
+        configureUI();
+    }
+
+    private void configureUI() {
+        _binding.groupBanner.setVisibility(View.VISIBLE);
+        _binding.btnDestroyAd.setVisibility(View.VISIBLE);
+        _binding.btnDestroyAd.setEnabled(false);
+        setTextColor(_binding.btnDestroyAd, R.color.lightGray);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:43]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        if (bannerAd != null) {
+            bannerAd.finishAd();
+            bannerAd.setAdListener(null);
+            bannerAd = null;
+        }
+    }
+
+    @Override
+    protected void playAd() {
+        super.playAd();
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:58]: another seemingly redundant override

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    @Override
+    public void onAdClicked(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbAdClick, R.color.black);
+
+    }
+
+    @Override
+    public void onAdEnd(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbAdEnd, R.color.black);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:85]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    @Override
+    public void onAdFailedToLoad(@NonNull BaseAd baseAd, @NonNull VungleError vungleError) {
+        setTextColor(_binding.lbError, R.color.black);
+        _binding.lbError.setText(vungleError.getErrorMessage());
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:92]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    @Override
+    public void onAdFailedToPlay(@NonNull BaseAd baseAd, @NonNull VungleError vungleError) {
+        setTextColor(_binding.lbError, R.color.black);
+        _binding.lbError.setText(vungleError.getErrorMessage());
+    }
+
+    @Override
+    public void onAdImpression(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbAdImpression, R.color.black);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:104]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    @Override
+    public void onAdLeftApplication(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbOnAdLeftApplication, R.color.black);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:110]: redundant line break

File: Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    @Override
+    public void onAdStart(@NonNull BaseAd baseAd) {
+        setTextColor(_binding.lbAdStart, R.color.black);
+        _binding.btnPlayAd.setEnabled(false);
+        setTextColor(_binding.btnPlayAd, R.color.lightGray);
+        _binding.btnDestroyAd.setEnabled(true);
+        setTextColor(_binding.btnDestroyAd, R.color.black);
+
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Java/app-v7-java/src/main/java/com/vungle/samples/samplejava/ui/fragment/InlineFragment.java:127]: redundant line break

File: Samples/android/Kotlin/app-v7-kotlin/src/main/java/com/vungle/samples/samplekotlin/MrecFragment.kt [hunk 11]
```diff
@@ -25,39 +24,32 @@ class MrecFragment : AdExperienceFragment(), BannerAdListener {
...
+        FrameLayout.LayoutParams.WRAP_CONTENT,
+        FrameLayout.LayoutParams.WRAP_CONTENT,
+        Gravity.CENTER_HORIZONTAL
+      )
+      binding.adContainer.addView(it, params)
     }
   }
 
   override fun playAd() {
     super.playAd()
```
Comment (anton-blashchuk-zoolatech) [inline Samples/android/Kotlin/app-v7-kotlin/src/main/java/com/vungle/samples/samplekotlin/MrecFragment.kt:41]: redundant override

**Review by anton-blashchuk-zoolatech — APPROVED** [review]
lgtm, except some nitpicks

**Review by ashinagawa — APPROVED** [review]
LGTM
- both Java and Kotlin sample worked without issues
- AppOpen, Banner , and MREC worked

@HuFangshuai InLine ads Placement will be updated later, right?

**Comment by HuFangshuai** [comment]
> LGTM
> 
> * both Java and Kotlin sample worked without issues
> * AppOpen, Banner , and MREC worked
> 
> @HuFangshuai InLine ads Placement will be updated later, right?

Yes.
